packages/relay/src/lib/eth.ts

```diff
@@ -850,26 +851,41 @@ export class EthImpl implements Eth {
   async getTransactionCount(address: string, blockNumOrTag: string, requestId?: string): Promise<string | JsonRpcError> {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     this.logger.trace(`${requestIdPrefix} getTransactionCount(address=${address}, blockNumOrTag=${blockNumOrTag})`);
-    const blockNumber = await this.translateBlockTag(blockNumOrTag, requestId);
-    if (blockNumber === 0) {
-      return EthImpl.zeroHex;
-    } else if (address && !blockNumOrTag) {
-      // get latest ethereumNonce
-      const mirrorAccount = await this.mirrorNodeClient.getAccount(address, requestId);
-      if (mirrorAccount && mirrorAccount.ethereum_nonce) {
-        return EthImpl.numberTo0x(mirrorAccount.ethereum_nonce);
-      }
-    }
 
-    // check consensus node as back up
     try {
-      const result = await this.mirrorNodeClient.resolveEntityType(address, requestId, [constants.TYPE_ACCOUNT, constants.TYPE_CONTRACT]);
-      if (result?.type === constants.TYPE_ACCOUNT) {
-        const accountInfo = await this.sdkClient.getAccountInfo(result?.entity.account, EthImpl.ethGetTransactionCount, requestId);
-        return EthImpl.numberTo0x(Number(accountInfo.ethereumNonce));
+      // cache considerations for high load
+      const cacheKey = `eth_getTransactionCount_${address}`;
+      let nonceCount = this.cache.get(cacheKey);
+      if (nonceCount) {
+        this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(nonceCount)}`);
+        return nonceCount;
       }
-      else if (result?.type === constants.TYPE_CONTRACT) {
-        return EthImpl.numberTo0x(1);
+
+      if (blockNumOrTag) {
+        const blockNum = Number(blockNumOrTag);
+        if (blockNum === 0) {
```

Comment:
Note previewnet and testnet have blocknumber = 1
Need to add consideration for this